src/test/groovy/com/jvm_bloggers/core/data_fetching/blogs/json_data/JsonBlogsDataFileValidationSpec.groovy

```diff
@@ -1,59 +1,51 @@
 import groovy.json.JsonSlurper
-import org.springframework.beans.factory.annotation.Value
-import org.springframework.boot.test.context.SpringBootTest
-import org.springframework.core.io.Resource
-import org.springframework.test.context.ActiveProfiles
-import org.springframework.test.context.ContextConfiguration
+import spock.lang.Shared
 import spock.lang.Specification
 import spock.lang.Subject
+import spock.lang.Unroll
 
-@ContextConfiguration
-@SpringBootTest(classes = JvmBloggersApplication)
-@ActiveProfiles("test")
 class JsonBlogsDataFileValidationSpec extends Specification {
 
-    @Value("classpath:blogs/bloggers.json")
-    Resource bloggersJson
+    @Subject
+    @Shared
+    List<BloggerEntry> jsonEntries
 
-    @Value("classpath:blogs/companies.json")
-    Resource companiesJson
+    private static final int TWITTER_USER_TAG_MAXIMUM_LENGTH = 16
 
-    @Value("classpath:blogs/videos.json")
-    Resource videosJson
+    private static final CharMatcher alphanumericOrUnderscore = (CharMatcher.inRange('a' as char, 'z' as char) as CharMatcher)
+            .or(CharMatcher.inRange('A' as char, 'Z' as char) as CharMatcher)
+            .or(CharMatcher.inRange('0' as char, '9' as char) as CharMatcher)
+            .or(CharMatcher.is('_' as char) as CharMatcher)
 
-    @Subject
-    List<BloggerEntry> jsonEntries
+    private static final CharMatcher lowercaseOrHyphen = (CharMatcher.inRange('a' as char, 'z' as char) as CharMatcher)
+            .or(CharMatcher.inRange('0' as char, '9' as char) as CharMatcher)
+            .or(CharMatcher.is('-' as char) as CharMatcher)
+
+    def setupSpec() {
+        File bloggersJson = new File('src/main/resources/blogs/bloggers.json')
+        File companiesJson = new File('src/main/resources/blogs/companies.json')
+        File videosJson = new File('src/main/resources/blogs/videos.json')
 
-    def setup() {
         JsonSlurper slurper = new JsonSlurper()
         jsonEntries = [bloggersJson, companiesJson, videosJson]
-            .collect {json -> json.inputStream}
             .collect({stream ->
                BloggersData data = slurper.parse(stream)
                data.getBloggers()
             })
             .flatten()
     }
 
-    def "should check that all bookmarkable IDs are lower-case, alphanumeric or hyphen"() {
-        List<String> violatingIds = []
-
-        when:
-        jsonEntries.bookmarkableId.forEach({id ->
-            CharMatcher lowercaseOrHyphen = (CharMatcher.inRange('a' as char, 'z' as char) as CharMatcher)
-                .or(CharMatcher.inRange('0' as char, '9' as char) as CharMatcher)
-                .or(CharMatcher.is('-' as char) as CharMatcher)
-            if (!lowercaseOrHyphen.matchesAllOf(id)) {
-                violatingIds.add(id)
-            }
-        })
+    @Unroll
+    def "should check that bookmarkable ID is lower-case, alphanumeric or hyphen"() {
+        expect:
+        lowercaseOrHyphen.matchesAllOf(id)
```

Comment:
When this test fails for one entry, it will be hard to say for which one. So maybe instead of `matchesAllOf` we could use @Unroll with #jsonEntry in test method name and test one entry in each run?

Reply:
Cool feature. Done.